### DISCUSSION TITLE
How to avoid allUsers in CloudRun invoker permission for corporate environment

### DISCUSSION THREAD

**Discussion by tszebeni-asml** [GoogleCloudPlatform/cloud-foundation-fabric#2425] [Q&A]
In a corporate environment usually (e.g. in our case) using allUsers is forbidden. In domain restricted sharing org policy we only allow identities belonging to our Cloud Identity to be assigned in iam policies.

How to create a Cloud Run service in a corporate environment, which:
- does not rely on allUsers
- uses only INTERNAL communication (egress internal)
- supports custom domain (ilb HTTPS)

Please advise, Thanks

Reference: https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/454a30b037dd267fada48cfd7e78cc86f85960f1/blueprints/serverless/cloud-run-corporate/main.tf#L140

**Comment by wiktorn** [comment]
If it uses ilb, will IAM grant to `domain:your-cloud-identity-domain-here.com` work for your case?

**Comment by ludoo** [comment]
Blueprints are meant as examples, you're expected to tweak them to suit your needs. :)
